iceoryx_utils/include/iceoryx_utils/internal/cxx/string.inl

```diff
@@ -265,6 +301,34 @@ inline string<Capacity>::operator std::string() const noexcept
     return std::string(c_str());
 }
 
+template <uint64_t Capacity>
+template <uint64_t N>
+inline string<Capacity>& string<Capacity>::copy(const string<N>& rhs) noexcept
+{
+    static_assert(N <= Capacity,
```

Comment:
I would perform here the self copy check
```
if ( this == &rhs ) {
...
```

Then you can remove the self copy check in the copy assignment operator and the copy method is safe to use.

Reply:
To avoid code duplication I call `copy` also in the copy constructors where this check is unnecessary.